src/mono/mono/metadata/threads-types.h

```diff
@@ -323,11 +321,11 @@ mono_threads_attach_coop (MonoDomain *domain, gpointer *dummy);
 MONO_API void
 mono_threads_detach_coop (gpointer cookie, gpointer *dummy);
 
-MonoDomain*
-mono_threads_attach_coop_internal (MonoDomain *domain, gpointer *cookie, MonoStackData *stackdata);
+void
+mono_threads_attach_coop_internal (gpointer *cookie, MonoStackData *stackdata);
```

Comment:
nit: attach and detach have callers in cominterop.c too

Reply:
But more importantly, there are callers in `marshal-ilgen.c` that use the return value.  those need to be adjusted.

Reply:
Ok, my mistake.  Those call the non-`_internal` versions. and I see you kept the same signatures for those.  Nevermind.